### ISSUE TITLE
Can't see iOS 11 simulators in OpenSim

### ISSUE THREAD

**Issue by hossamghareeb** [luosheng/OpenSim#37]
After updating my system to Hight Sierra and to Xcode 9.1. I can't see iOS 11 simulators like iPhone X in OpenSim list

**Comment by philosopherdog** [comment]
No longer working here either. Tried everything.

**Comment by luosheng** [comment]
Can you post the result of running `xcrun simctl list -j devices` in your terminal?

**Comment by hossamghareeb** [comment]
```
{
  "devices" : {
    "com.apple.CoreSimulator.SimRuntime.iOS-10-3" : [
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 5",
        "udid" : "806098BD-5435-40CC-B940-42D311F01465"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 5s",
        "udid" : "5B1C37F0-236E-4A41-BADE-AF328F8577B2"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 6",
        "udid" : "21BC2FD7-D496-4D2D-A8C5-88B9F8CAFC16"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 6 Plus",
        "udid" : "2EF9D112-5BD7-4CAA-83DA-C8FEF9F1C242"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 6s",
        "udid" : "F1EF6FCD-4524-48A5-A0F4-409312D0FA8F"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 6s Plus",
        "udid" : "C8B1576E-88F4-4C60-A71E-EB6DD93CE44C"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 7",
        "udid" : "873BEC5B-1D38-4DEB-A9B2-94F5C4E726F8"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 7 Plus",
        "udid" : "C11EC3DB-23E1-4109-A2F5-13CF409847F8"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone SE",
        "udid" : "B4A9255D-886B-4E8E-B51D-8F67926DB8E6"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPad Air",
        "udid" : "D68AC624-9EF0-4840-9CAB-0089B1037C6F"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPad Air 2",
        "udid" : "F979DC26-7B26-40E7-BB42-5278BF4C23E6"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPad (5th generation)",
        "udid" : "24C8FF20-A7DF-4F0C-8D2B-ED10E20F08F1"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPad Pro (9.7 inch)",
        "udid" : "950033B0-BEE1-4776-9742-3297D7F5CB79"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPad Pro (12.9 inch)",
        "udid" : "BFBBE5A1-1DCE-4316-BF09-6EFA64991406"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPad Pro (12.9-inch) (2nd generation)",
        "udid" : "B802FEB6-9A96-472D-8CF1-B1048B96953C"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPad Pro (10.5-inch)",
        "udid" : "491C4245-8781-4516-8AE1-F3E3CC62E1D4"
      }
    ],
    "iOS 11.1" : [
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 5s",
        "udid" : "B0A453C4-4ABA-4819-93BE-86B63DEF1D36"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 6",
        "udid" : "F2833856-9D07-446B-A5C6-3E92DCE08C7E"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 6 Plus",
        "udid" : "CDAE1E87-5F1E-4CE4-B2DE-C9E2316F312E"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 6s",
        "udid" : "C4257D42-B3ED-4764-BFF0-F42F5383AC1E"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 6s Plus",
        "udid" : "EBCCBCF9-1024-4F7C-BC7D-A96CCAC2E3AA"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 7",
        "udid" : "371886AA-E422-4681-AC4D-94D8BA0E58DE"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 7 Plus",
        "udid" : "9B22FD8D-D2EE-4070-823F-1C6CDC70BF23"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 8",
        "udid" : "ED97AB20-9589-4EFA-B640-BE5D78A87F09"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 8 Plus",
        "udid" : "435C48C2-7F5B-4173-B789-9EEF9B6F1204"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone SE",
        "udid" : "FC066C90-E5B9-4C1E-ACF0-AD3CE38AE179"
      },
      {
        "state" : "Booted",
        "availability" : "(available)",
        "name" : "iPhone X",
        "udid" : "EFB019E0-1F53-40DC-8333-DF95F61110CA"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPad Air",
        "udid" : "F1C26966-A3FE-41C1-BD49-1C989B0F0746"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPad Air 2",
        "udid" : "672E47DF-45B5-41A5-B876-0A82C53426AA"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPad (5th generation)",
        "udid" : "F7E3B3F2-144A-4660-BBA9-AC30074272FF"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPad Pro (9.7-inch)",
        "udid" : "BD08D942-47D4-46A7-8E5A-50D8069B9A19"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPad Pro (12.9-inch)",
        "udid" : "C9AE753D-7CC7-4EB8-8F22-7A9AC101EEB4"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPad Pro (12.9-inch) (2nd generation)",
        "udid" : "85D4394A-0C4B-45F2-A136-6210C515279F"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPad Pro (10.5-inch)",
        "udid" : "2D08F547-551F-496A-86ED-1F90ADF95C98"
      }
    ],
    "iOS 9.3" : [
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 4s",
        "udid" : "99155C43-C914-437F-B964-87DDC1761213"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 4s iOS 9.3",
        "udid" : "DE0820C5-A775-4735-9830-6C4AEB905B6C"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 5",
        "udid" : "F2B5A31C-8F75-493A-BC23-3F71F71DA677"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 5s",
        "udid" : "D64409F1-4D35-494E-B3F4-5BA54E49D285"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 6",
        "udid" : "7768AF01-384C-41DC-9051-852624E1E19A"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 6 Plus",
        "udid" : "9A8C8847-6486-4F11-8760-CD0D282FE890"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 6s",
        "udid" : "658D5829-BEC4-43B2-AE35-357CEC0637DC"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 6s Plus",
        "udid" : "FC4CD40C-D6BB-43D7-B4FE-9A0434D0B618"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPad 2",
        "udid" : "94A9D14E-01A7-463F-98A1-AFDA2A301132"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPad Retina",
        "udid" : "A72514C2-ADD2-4AC0-8ACE-A10915BD2686"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPad Air",
        "udid" : "384690AA-B114-49D1-B082-FA91D2679385"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPad Air 2",
        "udid" : "D7DF1E83-0C47-4FC4-9FE7-E29F993E3C61"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPad Pro",
        "udid" : "A48A477A-A523-4471-9ADE-516A50173DB1"
      }
    ],
    "com.apple.CoreSimulator.SimRuntime.watchOS-3-2" : [
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "Apple Watch - 38mm",
        "udid" : "F678D86B-7A6B-4A08-9F94-F6C443ADC47E"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "Apple Watch - 42mm",
        "udid" : "6F8A02DA-F6AD-4BE0-B938-95A1CDE40FDD"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "Apple Watch Series 2 - 38mm",
        "udid" : "FC1987A5-E8E6-43A6-9231-CB0356416191"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "Apple Watch Series 2 - 42mm",
        "udid" : "34FAC454-DA81-414C-B8ED-910E04BC2743"
      }
    ],
    "com.apple.CoreSimulator.SimRuntime.tvOS-10-0" : [
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "Apple TV 1080p",
        "udid" : "D359C210-9D9F-437A-B3F3-F1EE3011CD68"
      }
    ],
    "com.apple.CoreSimulator.SimRuntime.watchOS-3-1" : [
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "Apple Watch - 38mm",
        "udid" : "9D5ED6C8-E93F-42AE-9F83-764E56F693BE"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "Apple Watch - 42mm",
        "udid" : "4CBF978D-8364-43F4-A9B5-B08FE96F54F5"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "Apple Watch Series 2 - 38mm",
        "udid" : "EA3A4147-3F36-4D39-A275-23EDCB74FA69"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "Apple Watch Series 2 - 42mm",
        "udid" : "369A96DD-B55D-420B-B94B-086204B565EC"
      }
    ],
    "com.apple.CoreSimulator.SimRuntime.tvOS-10-1" : [
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "Apple TV 1080p",
        "udid" : "8ED3AD17-B983-4B52-97C3-E08E510B9D58"
      }
    ],
    "watchOS 4.1" : [
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "Apple Watch - 38mm",
        "udid" : "F35AC297-B361-400A-8F75-5FFA2EB7CCDB"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "Apple Watch - 42mm",
        "udid" : "92EAF9D3-D146-49E6-A3B5-087117DC0D1E"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "Apple Watch Series 2 - 38mm",
        "udid" : "FE2FACB5-A408-463E-8112-578BFA15F5CB"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "Apple Watch Series 2 - 42mm",
        "udid" : "AE2D4D98-D5D2-4EBD-BF32-5BFFEB8E8F17"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "Apple Watch Series 3 - 38mm",
        "udid" : "3D17BE1A-EEC3-4865-9E46-96D31A46E0B0"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "Apple Watch Series 3 - 42mm",
        "udid" : "40711B3E-78B3-48B6-82DB-342190DD2642"
      }
    ],
    "com.apple.CoreSimulator.SimRuntime.tvOS-10-2" : [
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "Apple TV 1080p",
        "udid" : "E8B15E13-F68B-412B-938C-70EF4153C8EE"
      }
    ],
    "com.apple.CoreSimulator.SimRuntime.iOS-10-1" : [
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 5",
        "udid" : "DA21E1B8-869E-430A-9280-639212E1554F"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 5s",
        "udid" : "960A50D6-507A-48A5-8D7E-2BEC8AD7A3B8"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 6",
        "udid" : "0BE09B89-8838-412D-8CA1-7791DDE41EAF"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 6 Plus",
        "udid" : "30733699-E31C-4681-850A-D47DF66FAC35"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 6s",
        "udid" : "0095CA63-6743-426B-8149-15149130FF4D"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 6s Plus",
        "udid" : "8F0D7DB0-AD3C-4D96-87E0-5993412E520C"
      },
      {
        "state" : "Creating",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 7",
        "udid" : "26C6BEFA-9007-4618-BA6D-9716A8FB22F0"
      },
      {
        "state" : "Creating",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 7 Plus",
        "udid" : "9F1ACD61-CD93-4F5A-A6E3-395693B68C07"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone SE",
        "udid" : "E601AA2D-1B78-4335-8CC4-F647C133478A"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPad Retina",
        "udid" : "C1397446-D735-4A6D-A0D5-CC6F6250DC2E"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPad Air",
        "udid" : "90C4617A-662B-4E21-A201-18B6A73BF28B"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPad Air 2",
        "udid" : "A88FA81B-6491-4D78-AAD8-9A06FEA02AFA"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPad Pro (9.7 inch)",
        "udid" : "27D84C90-CEED-42EF-8064-5C81A3E91EBE"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPad Pro (12.9 inch)",
        "udid" : "201787B0-2FA0-480E-A56D-E7E639E59447"
      }
    ],
    "tvOS 11.1" : [
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "Apple TV",
        "udid" : "3EDD5A41-F547-4B27-B995-CEB7865506B7"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "Apple TV 4K",
        "udid" : "6DEB0EC6-EA1B-4986-9290-3BF281C5FDDD"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "Apple TV 4K (at 1080p)",
        "udid" : "F027D8AE-4ADC-4320-B645-A6530A943614"
      }
    ],
    "com.apple.CoreSimulator.SimRuntime.iOS-10-2" : [
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 5",
        "udid" : "D6FBA676-B378-422D-B19D-4AFB69188CA7"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 5s",
        "udid" : "1370BB7B-037D-4871-99B0-3EBD345F5274"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 6",
        "udid" : "DCD21D45-B50C-4CDC-A3DC-7F1AA3CAA348"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 6 Plus",
        "udid" : "4A3C39D8-B5A1-406B-B0C5-BCD8B57F0C55"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 6s",
        "udid" : "BDB49B8F-25E2-4959-87A1-87AB8926DAB7"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 6s Plus",
        "udid" : "F05B7702-8EF9-4D4B-B522-76FC10C6FC8D"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 7",
        "udid" : "A672C1F0-AF6D-4ED2-B1E5-B18F7C144E58"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone 7 Plus",
        "udid" : "EF063016-E5F7-403D-8002-FC7B2E986DDD"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPhone SE",
        "udid" : "D78423F3-AF82-4DB8-8981-6646C6E23794"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPad Retina",
        "udid" : "B17DDC13-F95D-4F3D-BD16-43EC70C100D2"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPad Air",
        "udid" : "DCE4DE9A-1BD7-45F7-8FC1-5F475022975F"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPad Air 2",
        "udid" : "D14845D2-7876-4030-9985-A430CE8876FC"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPad Pro (9.7 inch)",
        "udid" : "1577E512-6DF2-4001-84E2-4616C08C57E8"
      },
      {
        "state" : "Shutdown",
        "availability" : " (unavailable, runtime profile not found)",
        "name" : "iPad Pro (12.9 inch)",
        "udid" : "7E9E7366-CF5A-4589-B8A0-E7802BB88CB3"
      }
    ],
    "iOS 8.4" : [
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 4s",
        "udid" : "08EED355-5E8A-499E-A3C5-63EBFED752B1"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 5",
        "udid" : "B2EF76A6-A13A-4B37-A317-AE3B667A6795"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 5s",
        "udid" : "F2EA9C45-5436-46F4-AB03-319CDF734571"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 6",
        "udid" : "A561E359-15F2-45AF-AD21-9526D431AB3B"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPhone 6 Plus",
        "udid" : "46C5D8FB-C000-4A41-938C-BB628CF1F7C7"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPad 2",
        "udid" : "3692A79B-B880-4279-9AB9-32B20C679615"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPad Retina",
        "udid" : "661323CE-BFD1-4736-B6A2-66ACA70CA8AA"
      },
      {
        "state" : "Shutdown",
        "availability" : "(available)",
        "name" : "iPad Air",
        "udid" : "3093B41A-B75E-427F-A97B-7F561354A4CB"
      }
    ]
  }
}
```

**Comment by basilmariano** [comment]
same here :( same issue

**Comment by hossamghareeb** [comment]
I just restarted my MacBook and all of a sudden it works now and I can see all simulators. Not sure what was the issue

**Comment by luosheng** [comment]
@hossamghareeb Thanks for the feedback!

**Comment by basilmariano** [comment]
hmm I can see some of the simulators but some the application inside it are not visible. anyone who encounter this issue?

**Comment by VincentSit** [comment]
For me, recompiling from source code can solve this problem, you can try it or [download](https://www.dropbox.com/s/z6qo18zdf9af863/OpenSim.app.zip?dl=1) my export directly.

**Comment by davidnuvolar** [comment]
A colleague using High Sierra had the same issue. @VincentSit build worked for him, although.

**Comment by vemulanishanth** [comment]
I tried to remove Xamarin, xcode and all the development utilities before reinstalling xcode again in vain.
@VincentSit thanks for the exported build but it didn't work. recompiling the source code did not work either.
anyone else with issues not solved yet?
I'm installing iOS 11.1 and 9.3 simulators to check if the issue is with only preinstalled simulators or all of them.
I'm up to use all the help I can get.

**Comment by vemulanishanth** [comment]
After re-installing xcode I did even run any application at least once before running OpenSim. Once I run an iOS app under development and then running OpenSim again there are no more issues.

**Comment by kryzhikh** [comment]
Doesn't show any simulator on 10.13.3, Xcode 9.2
Reboot doesn't help.
`xcrun simctl list -j devices` shows all my sims

Although works fine on another mac with the same macOS and Xcode versions.

**Comment by KimPark89** [comment]
@luosheng I discovered the problem.
In iOS11 there are 2 files inside the _/Bundle/Application/_ folder (_example_.app and BundleMetadata.plist). 
So when you call the _.last_ method to get the .app file inside the contents url array (in Application.swift line 38) sometimes you get the wrong file.
You can fix this checking the file extension:

```
guard let contents = try? FileManager.default.contentsOfDirectory(at: url, includingPropertiesForKeys: nil, options: [.skipsSubdirectoryDescendants, .skipsHiddenFiles]),
            let url = contents.filter({ (url) -> Bool in return url.lastPathComponent.hasSuffix(".app") }).last // url ".app" diretory
            else {
                return nil
        }
```

**Comment by luosheng** [comment]
@KimPark89 Thanks for the information, but I think it was fixed in e1cc71a. Can you let me know which version of code are you referring to?

**Comment by KimPark89** [comment]
You right @luosheng. I was using the release 0.3.0. Sorry.

**Comment by kryzhikh** [comment]
Hi @luosheng 
So looks like the latest release doesn't contain the fix you've mentioned. Are you going to issue a new release any time soon?

**Comment by christian-schneider** [comment]
It might be that your /Applications/Xcode.app/Contents/Developer/usr/bin/ is not on the path, the directory where simctl lives, i just had this on High Sierra with Xcode 9.2. It might have happened when I installed the command line tools manually from the download. It sometimes did work when running directly out of Xcode, but never when started manually. It turned out that simctl could not be found when I tried to run 'xcrun simctl list -j devices'. I added the above mentioned path to my .zshrc and now it works! Worth checking out on your machine if you still have the problem. @luosheng, maybe you want to pack the required binary with the executable?

**Comment by Nathan187** [comment]
i too am having problems with not all the apps showing for a particular simulator.  hoping for a fix soon

**Comment by cjwhitsitt** [comment]
I have this issue with Xcode 9.2 and 9.3. I am able to run `xcrun simctl list -j devices` successfully and it shows my iOS 11.0-11.3 devices.

**Comment by andriichernenko** [comment]
Cloning the repo and building the app from the source code fixed this issue for me.

**Comment by cjwhitsitt** [comment]
Any idea when the next version will be release to include e1cc71a? I realize the workaround until then is to clone and build from source.

**Comment by basilmariano** [comment]
I install OpenSims and when i open it I only see 3 submenu items from its menu item

<img width="201" alt="screen shot 2018-08-25 at 4 34 00 am" src="https://user-images.githubusercontent.com/5466582/44606614-47b0f780-a820-11e8-906d-264fb8e60c21.png">

anyone?

**Comment by Nathan187** [comment]
download the source code and run it in a different instance of xcode.  should work...that's what i do at least

**Comment by martheli** [comment]
I can see one of my simulators but not all. I am currently running a project on an iOS running iOS 11 but it is not showing:

<img width="185" alt="screen shot 2018-08-30 at 9 16 56 pm" src="https://user-images.githubusercontent.com/20176169/44887452-4fe2ba00-ac9a-11e8-8b66-51d4f7d87440.png">

**Comment by joachimVdH** [comment]
I just installed opensim through brew cask, xcode 9.4.1, osx 10.13.6. 
No simulators are shown in the menuapp. 
```
xcrun simctl list -j devices
xcrun: error: unable to find utility "simctl", not a developer tool or in PATH
```

**Comment by joachimVdH** [comment]
`xcrun simctl list -j devices` lists all devices and ios since I selected the 'Command Line Tools' version  in  Xcode > Preferences > Locations 

but still no simulators in the app.

**Comment by luosheng** [comment]
This was fixed in e1cc71a00edc987821596a44fdbedd1ac2af2b31 but I forgot to upload a new build...

A new build 0.4.0 has been released and uploaded. Please check out if any issue remains.
